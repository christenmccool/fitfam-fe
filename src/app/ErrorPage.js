import React from 'react';

import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

/** Error page */
const ErrorPage = ({ errors }) => {
  return (
    <Container maxWidth="sm" align="center" sx={{backgroundColor: "#FFF", borderRadius: '10px'}}>
      <Box my={4} p={5}>
        {Array.isArray(errors) ?
          errors.map(error => (
            <Typography key={error} variant="h6">
              {error}
            </Typography>
          )
        )
        : 
        <Typography variant="h6">
          Server error. Try again later.
        </Typography>
      }

      </Box>
    </Container>
  )
}

export default ErrorPage;