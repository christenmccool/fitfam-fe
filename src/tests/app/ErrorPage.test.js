import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ErrorPage from "../../app/ErrorPage";


it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <ErrorPage />
    </MemoryRouter>
  );
});


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ErrorPage />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders text if errors passed in", function () {
  const errors = ["User must be logged in to access this page", "Only admin can change family name"]
  render(
    <MemoryRouter>
      <ErrorPage errors={errors} />
    </MemoryRouter>
  );
  expect(screen.getByText("User must be logged in to access this page")).toBeInTheDocument();
  expect(screen.getByText("Only admin can change family name")).toBeInTheDocument();
});


it("renders text if no errors passed in", function () {
  render(
    <MemoryRouter>
      <ErrorPage />
    </MemoryRouter>
  );
  expect(screen.getByText("Server error. Try again later.")).toBeInTheDocument();
});



