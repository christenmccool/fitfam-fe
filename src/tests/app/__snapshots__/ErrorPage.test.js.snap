// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root MuiContainer-maxWidthSm css-ead4d9-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-19rm58x"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Server error. Try again later.
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
